### PR TITLE
fix(clickhouse): Validate decoded trace/span ID lengths to prevent panic on corrupted rows

### LINKED ISSUE #8933: [Bug]: ClickHouse v2 storage panics on empty or short TraceIDs/SpanIDs during row conversion
### What happened?

While reviewing the ClickHouse v2 storage implementation, I found a panic in the read path.

In `internal/storage/v2/clickhouse/tracestore/dbmodel/from.go`, the `convertSpan` function uses `hex.DecodeString` to read `sr.TraceID` and `sr.ID` into byte slices. It immediately casts these slices to `pcommon.TraceID` and `pcommon.SpanID` (which are aliases for `[16]byte` and `[8]byte` arrays):

```go
traceId, err := hex.DecodeString(sr.TraceID) // returns []byte
// ...
span.SetTraceID(pcommon.TraceID(traceId)) // Panics if len(traceId) < 16
```

If `sr.TraceID` is empty (length 0) or shorter than 32 hex characters, `hex.DecodeString` returns a slice shorter than 16 bytes with no error. However, casting a short slice to a fixed `[16]byte` array causes a runtime panic in Go.

Because this happens on the read path, a single corrupted row or empty `TraceID` in ClickHouse will completely crash the `jaeger-query` (or all-in-one) process when it attempts to load that trace.

### Steps to reproduce

1. Run Jaeger with the ClickHouse v2 storage backend.
2. Manually insert a single row into the ClickHouse `spans` table with an empty `trace_id` (or one shorter than 32 hex chars).
3. Call the `GetTrace` or `FindTraces` API that attempts to read that row.
4. Observe that the Jaeger process crashes with a slice-to-array out-of-bounds panic.

### Expected behavior

Jaeger should gracefully handle malformed data from the storage layer. It should either drop the corrupted span
with a warning (using `jptrace.AddWarnings`) or safely pad/sanitize the ID, rather than allowing a runtime panic to crash the entire service.

### Relevant log output

```shell
panic: runtime error: cannot convert slice with length 0 to array or pointer to array with length 16
    
goroutine 123 [running]:
github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel.convertSpan(...)
/go/src/github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go:76
github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel.FromRow(...)
/go/src/github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go:28
```

### Screenshot

_No response_

### Additional context

The fix is straightforward. We just need to check the length of the decoded slice before casting, and either copy
it into a fixed  [16]byte  buffer safely or return an error/warning. 

Happy to open a PR to fix this for `TraceID`, `SpanID`, `ParentSpanID`, and `Link` IDs in `from.go`. 

### Jaeger backend version

v2 (latest)

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

ClickHouse v2

### Operating system

Linux / macOS

### Deployment model

Local development / Docker

### Deployment configs

```shell

```

### PR DESCRIPTION
**Pull request by iammdzaidalam**
## Which problem is this PR solving?

- Resolves #8933.
- A malformed or truncated trace/span ID stored in ClickHouse can trigger a slice-to-array conversion panic while decoding spans.
- This allows a single corrupted row to act as a "poison pill", crashing `jaeger-query` whenever the trace is read instead of handling the corrupted data gracefully.
## Description of the changes

- Added robust `decodeTraceID` and `decodeSpanID` helper functions in `internal/storage/v2/clickhouse/tracestore/dbmodel/from.go`.
- Replaced bare `hex.DecodeString` slice-to-array type casts (e.g. `pcommon.TraceID(traceId)`) with safe byte-copying that validates length first.
- Fixed `TraceID`, `SpanID`, `ParentSpanID`, and `Link` ID decoding logic to gracefully return an error/warning instead of triggering an out-of-bounds array pointer runtime panic.

## How was this change tested?

- Added exhaustive unit test coverage in `from_test.go` checking all edge cases (empty strings, too short, too long) for `TraceID`, `SpanID`, `ParentSpanID`, and `Link` IDs.
- Added `require.NotPanics` wrappers in tests to definitively prove the runtime panic has been prevented.

## Checklist

- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully: `make lint test`

## AI Usage in this PR (choose one)

- [x] **Light**: AI provided minor assistance (formatting, simple suggestions)

### CODE REVIEW

File: internal/storage/v2/clickhouse/tracestore/dbmodel/from.go [hunk 1]
```diff
@@ -108,18 +138,18 @@ func convertSpan(sr *SpanRow) (ptrace.Span, error) {
...
-		traceID, err := hex.DecodeString(l)
+		traceID, err := decodeTraceID(l)
 		if err != nil {
 			jptrace.AddWarnings(span, fmt.Sprintf("failed to decode link trace ID: %v", err))
 			continue
 		}
-		link.SetTraceID(pcommon.TraceID(traceID))
-		spanID, err := hex.DecodeString(sr.LinkSpanIDs[i])
+		link.SetTraceID(traceID)
+		spanID, err := decodeSpanID(sr.LinkSpanIDs[i])
```
[the remark this answers is not in the document]
Comment (iammdzaidalam) [reply internal/storage/v2/clickhouse/tracestore/dbmodel/from.go:147]: Well nice catch from a pure Go struct perspective... but this is protected natively at the database schema layer.

In `create_spans_table.sql`, `links` is defined as a ClickHouse `Nested` data structure. The ClickHouse engine strictly guarantees that all sub-columns of a `Nested` column are perfectly parallel arrays of the exact same length. It is physically impossible for the database to return a row where `LinkSpanIDs` has a different length than `LinkTraceIDs`.

Because `SpanRow` is mapped directly from these guaranteed-parallel arrays by the `clickhouse-go` driver, this out-of-bounds index cannot be triggered in production.

btw happy to add a defensive `if i >= len(...) { break }` bounds check though, iff maintainers want to bulletproof this against manual struct initialization in future unit tests.

**Review by yurishkuro — APPROVED** [review]
